src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -1065,6 +1067,54 @@ public function testMirrorContentsWithSameNameAsSourceOrTargetWithDeleteOption()
         $this->assertFileNotExists($targetPath.'target');
     }
 
+    public function testMirrorWithIterator()
+    {
+        $sourcePath = $this->workspace.DIRECTORY_SEPARATOR.'source'.DIRECTORY_SEPARATOR;
+
+        mkdir($sourcePath);
+        mkdir($sourcePath.'source1');
+        mkdir($sourcePath.'source2');
+
+        $targetPath = $this->workspace.DIRECTORY_SEPARATOR.'target'.DIRECTORY_SEPARATOR;
+
+        // Mirror everything except `source2`
+        $iterator = new Finder;
+        $iterator->in($sourcePath)
+            ->exclude('source2');
+        $this->filesystem->mirror($sourcePath, $targetPath, $iterator);
+
+        $this->assertTrue(is_dir($targetPath));
+        $this->assertDirectoryExists($targetPath.'source1');
+        $this->assertDirectoryNotExists($targetPath.'source2');
+    }
```

Comment:
This test works fine, I only wrote it because the `$iterator` parameter is not covered at all by tests.